core/src/main/java/com/linecorp/armeria/client/DefaultClientRequestContext.java

```diff
@@ -282,11 +289,24 @@ private DefaultClientRequestContext(DefaultClientRequestContext ctx,
         responseTimeoutMillis = ctx.responseTimeoutMillis();
         maxResponseLength = ctx.maxResponseLength();
         additionalRequestHeaders = ctx.additionalRequestHeaders();
+        customizers = copyThreadLocalCustomizers();
 
         for (final Iterator<Attribute<?>> i = ctx.attrs(); i.hasNext();) {
             addAttr(i.next());
         }
-        runThreadLocalContextCustomizer();
+
+        runThreadLocalContextCustomizers();
```

Comment:
How about adding a small example that uses `RetryingClient` so that we can be sure that it works as expected?

Reply:
Will do!

Reply:
What's the expected behavior by the way when `RetryingClient` is involved?

Reply:
I realized that we should not invoke the customizers for derived contexts. Just removed the call. Less code = no need to test. :rofl: